data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineParser.java

```diff
@@ -187,11 +191,22 @@ private void buildPipelineFromConfiguration(
 
             final List<Buffer> secondaryBuffers = getSecondaryBuffers();
             LOG.info("Constructing MultiBufferDecorator with [{}] secondary buffers for pipeline [{}]", secondaryBuffers.size(), pipelineName);
-            final MultiBufferDecorator multiBufferDecorator = new MultiBufferDecorator(buffer, secondaryBuffers);
+            final MultiBufferDecorator multiBufferDecorator = new MultiBufferDecorator(pipelineDefinedBuffer, secondaryBuffers);
+
+
+            final Buffer buffer;
+            if(source instanceof PipelineConnector) {
+                buffer = multiBufferDecorator;
```

Comment:
Could we use the CircuitBreakingBuffer here with a higher usage threshold? Or perhaps use a slightly lower usage threshold than specified for non-PipelineConnector sources. Slightly offsetting these should avoid deadlock while preventing OOM. Though performance would take a hit

Reply:
That might be a good follow-on.

Currently, the `PipelineConnector` shuts down the whole pipeline if it fails to write to the buffer. So supporting this might be a slightly larger change.